### PR TITLE
Documentation: Update Truth Data instructions

### LINKED ISSUE #2068: Internal: Update Contributor's Guide to use GH Action to update truth data
Currently the chapter for adding new use cases include instructions to resolve conflicts between the develop and develop-ref branches, then create a PR to merge changes from develop-ref into develop that triggers a workflow to update the truth data.

## Describe the Task ##
Update the instructions to manually trigger the "Update Truth Data" workflow to update the truth data. Consider including screenshots.

* Navigate to https://github.com/dtcenter/METplus/actions/workflows/update_truth.yml
* Click on Run Workflow button on right
* Select the branch to update (develop or main_vX.Y only)
* Enter pull request(s) that introduced the difference. PRs from this repo can be formatted \#XXXX and PRs from other repos can be formatted {repo}\#XXXX e.g. `dtcenter/MET#2475`
* Enter a short description of the differences
* Click the green "Run Workflow" button. When the workflow has completed, a new pull request should have been created.
* Navigate to the Pull Request tab and click on the new pull request.
* Review the content and adjust as needed
* Merge the pull request (do not need to wait for checks to complete)
* Navigate to Actions tab and confirm that -ref branch workflow was started
* Confirm run completes successfully and truth data was updated

### Time Estimate ###
~4 hours

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
None

### Funding Source ###
None

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [ ] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issues
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by georgemccabe**
Updated contrib guide instructions to use GitHub Actions workflow to update truth data instead of doing it manually

## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Review documentation changes: https://metplus.readthedocs.io/en/feature_doc_update_truth/Contributors_Guide/add_use_case.html#update-the-truth-data

- [X] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [X] Do these changes include sufficient testing updates? **[Yes]**

- [X] Will this PR result in changes to the test suite? **[No]**</br>

- [X] Please complete this pull request review by **5/5/2023**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [X] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [X] Review the source issue metadata (required labels, projects, and milestone).
- [X] Complete the PR definition above.
- [X] Ensure the PR title matches the feature or bugfix branch name.
- [X] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [X] After submitting the PR, select **Development** issue with the original issue number. **no issue!**
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/Contributors_Guide/add_use_case.rst [hunk 1]
```diff
@@ -1082,46 +1083,41 @@ the destination branch. This is done so that the pull request number
...
+* Navigate to https://github.com/dtcenter/METplus/actions/workflows/update_truth.yml
+  or from the METplus GitHub page, click on the Actions tab,
+  then click on "Update Truth Data" under menu on the left.
+* Click on the "Run workflow" button on the right.
+* Click on the Branch pull down and select "develop" unless you are updating
+  the truth data for a bugfix on a main_vX.Y branch.
+* Enter the pull request numbers that warranted the update.
+  Include the '#' symbol before the number to create a link to the PR.
+  PRs from a repository other than METplus should include
+  the repository name before '#' symbol.
```
Comment (JohnHalleyGotway) [inline docs/Contributors_Guide/add_use_case.rst:1105]:
```suggestion
  'dtcenter/' and the repository name before '#' symbol (e.g. dtcenter/MET#2000).
```

File: docs/Contributors_Guide/add_use_case.rst [hunk 2]
```diff
@@ -1082,46 +1083,41 @@ the destination branch. This is done so that the pull request number
...
+
+* Click the "Run workflow" button.
+* A new workflow run should appear at the top of the list and complete quickly.
+* Click on the "Pull Requests" tab.
+  A new pull request should have been created with the information that
+  was entered. Click on the new pull request.
+* Verify that the information in this pull request is correct.
+  If the "develop" branch was selected in the "Run workflow" menu,
+  then the pull request should show **develop-ref <- develop**.
 * Add the appropriate project and milestone values on the right hand side.
```
Comment (JohnHalleyGotway) [inline docs/Contributors_Guide/add_use_case.rst:1119]:
```suggestion
* Add the appropriate project, development cycle, and milestone values on the right hand side.
```

**Review by JohnHalleyGotway — APPROVED** [review]
Thanks for adding these details. I read through the updates and proposed 2 changes that I'd recommend including.

With those tweaks, I approve of the this PR.

Have a nice weekend.

**Comment by georgemccabe** [comment]
After I created this pull request, I discovered that a step was missed to update the sample input data used for the automated tests from another pull request. Before merging this PR, I plan to make the suggested edits from @JohnHalleyGotway and add a step to ensure that the missing step is checked before triggering the GitHub Actions run to update the truth data.

File: docs/Contributors_Guide/add_use_case.rst [hunk 3]
```diff
@@ -1071,8 +1073,9 @@ then combine the two list items into a single item::
 
     - "met_tool_wrapper:0-54"
 
-Create a pull request from develop into develop-ref
----------------------------------------------------
+
+Update the Truth Data
+---------------------
 
 The addition of a new use case results in new output data. When this happens,
```
Comment (JohnHalleyGotway) [inline docs/Contributors_Guide/add_use_case.rst:1080]: George, I realize that this section is about adding a new use case. But I do think its worth noting in some way that changes to existing use cases and bugfixes can also change the output. Feel free to use the wording below... or re-word it however you see fit.

```suggestion
The addition of a new use case, changes to an existing use case, or software bug fixes can result in new or modified output data. When this happens,
```

**Review by JohnHalleyGotway — APPROVED** [review]
George, I think the note to confirm that the new use case has sufficient input data to run is a good idea. It might also to be wise to add information about what failure looks like. If you fail to add the input data or put it in the wrong spot, what would that failure look like.

Overall, I approve of these changes, but please address my previous comments.

**Pull request merged**